src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_WebView.cs

```diff
@@ -1,48 +1,118 @@
 ﻿using System;
+using System.Threading.Tasks;
+using Private.Infrastructure;
 using Windows.UI.Xaml.Controls;
 
-namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls
+namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls;
+
+#if !HAS_UNO || __ANDROID__ || __IOS__ || __MACOS__
+[TestClass]
+[RunsOnUIThread]
+public class Given_WebView
 {
-	[TestClass]
-	[RunsOnUIThread]
-	public class Given_WebView
+	[TestMethod]
+	public void When_Navigate()
 	{
+		var webView = new WebView();
+		var uri = new Uri("https://bing.com");
+		webView.Navigate(uri);
+		Assert.IsNotNull(webView.Source);
+		Assert.AreEqual("https://bing.com/", webView.Source.OriginalString);
+		Assert.AreEqual("https://bing.com", uri.OriginalString);
+	}
+
 #if __ANDROID__ || __IOS__ || __MACOS__
```

Comment:
I assume this is needed because `NavigateWithHttpRequestMessage` has wrong signature currently?

Consider adding a comment clarifying that.

Once this is merged, the next rebase to breaking-changes branch is going to have a conflict and the comment would be good to help correctly remove this when resolving the conflicts.

FYI @jeromelaban

Reply:
I am not too sure if this overload is actually useful, as in the end it just uses the request URI anyway, but kept it for legacy compatibility. Can be addressed in the future.